### PR TITLE
deps: Bump `com_envoyproxy_protoc_gen_validate` -> 0.6.12

### LINKED ISSUE #23357: Newer release available `com_envoyproxy_protoc_gen_validate`: v0.6.13 (current: v0.6.7)
Package Name: com_envoyproxy_protoc_gen_validate@0.6.7
Current Version: v0.6.7@2022-03-04
Available Version: v0.6.13@2022-10-03
Upstream releases: https://github.com/envoyproxy/protoc-gen-validate/releases

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Fix #23357 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/bazel/repository_locations.bzl [hunk 1]
```diff
@@ -16,9 +16,9 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         project_desc = "protoc plugin to generate polyglot message validators",
         project_url = "https://github.com/envoyproxy/protoc-gen-validate",
-        version = "0.6.7",
-        sha256 = "4c692c62e16c168049bca2b2972b0a25222870cf53e61be30b50d761e58728bd",
-        release_date = "2022-03-04",
+        version = "0.6.13",
+        sha256 = "cfa8741c939387969550a305f237e627db26e3ca00c69c0d6a5148843d14655a",
+        release_date = "2022-10-03",
         strip_prefix = "protoc-gen-validate-{version}",
         urls = ["https://github.com/envoyproxy/protoc-gen-validate/archive/v{version}.tar.gz"],
```
Comment (moderation) [inline api/bazel/repository_locations.bzl:23]: Change to
```
        urls = ["https://github.com/bufbuild/protoc-gen-validate/archive/v{version}.tar.gz"],
```

File: api/bazel/repository_locations.bzl [hunk 2]
```diff
@@ -16,9 +16,9 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "protoc-gen-validate (PGV)",
         project_desc = "protoc plugin to generate polyglot message validators",
         project_url = "https://github.com/envoyproxy/protoc-gen-validate",
```
Comment (moderation) [inline api/bazel/repository_locations.bzl:18]: `https://github.com/bufbuild/protoc-gen-validate`

**Review by moderation — CHANGES_REQUESTED** [review]
With the transition of PGV to Buf, we'll need to update all the references - https://github.com/envoyproxy/envoy/search?q=%22github.com%2Fenvoyproxy%2Fprotoc-gen-validate%22&type=code

**Comment by RyanTheOptimist** [comment]
@moderation's suggestions seem good
/wait

File: api/bazel/repository_locations.bzl (hunk 2 above)
Comment (phlax) [reply api/bazel/repository_locations.bzl:18]: seems like there are a few more refs beyond the deps hookup - ive opened a ticket  #23424

**Review by wbpcode — APPROVED** [review]
lgtm. Thanks. 😀

**Pull request merged**